### PR TITLE
JDP230402-17

### PR DESCRIPTION
**Pull request by WitoZak**
Added ProductEntityTest, OrderItemRepository method deleted, Cart & Group tiny changes and added dependency for Junit5

### CODE REVIEW

**Comment by WitoZak** [comment]
Ok, so let's start with the tests :) Be critical, I don't know if I did it correctly. As far as I remember, we haven't done entity tests without using mapper and service classes yet at the bootcamp. I think I have included all the relationships to the Product entity and tested it as best as I could. The tests have passed successfully. I had to add a dependency to use Junit5. I'm waiting for your comments.

File: build.gradle [hunk 1]
```diff
@@ -18,11 +18,12 @@ dependencies {
...
     runtimeOnly 'com.h2database:h2'
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.9.1'
     runtimeOnly 'com.mysql:mysql-connector-j'
     compileOnly 'org.projectlombok:lombok'
     annotationProcessor 'org.projectlombok:lombok'
 }
 
-tasks.named('test') {
+test {
```
Comment (mwojtyla) [inline build.gradle]: If I remember correctly you wanted to change it once ;) Do you really need it? It's not important and it will be better to focus on other things than resolve later conflicts in merge with such things.

Of course the new dependence is correct and needed :)

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 2]
```diff
@@ -9,11 +9,12 @@
 import javax.persistence.*;
 import java.util.List;
 
-@NoArgsConstructor
+@NoArgsConstructor(force = true)
 @AllArgsConstructor
```
Comment (mwojtyla) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: Why did you change this annotation for `@NoArgsConstructor`?

File: src/main/java/com/kodilla/ecommercee/domain/Group.java [hunk 3]
```diff
@@ -11,9 +11,9 @@
 import java.util.List;
 
 @Entity
-@Table (name = "GROUPS")
+@Table (name = "PRODUCT_GROUP")
 @AllArgsConstructor
-@NoArgsConstructor
+@NoArgsConstructor(force = true)
```
Comment (mwojtyla) [inline src/main/java/com/kodilla/ecommercee/domain/Group.java]: Why did you change this annotation for `@NoArgsConstructor`?

File: src/main/java/com/kodilla/ecommercee/repository/OrderItemRepository.java [hunk 4]
```diff
@@ -3,18 +3,11 @@
...
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
+
 import javax.transaction.Transactional;
-import java.util.List;
-import java.util.Optional;
 
 @Transactional
 @Repository
 public interface OrderItemRepository extends CrudRepository<OrderItem, Long> {
```
Comment (mwojtyla) [inline src/main/java/com/kodilla/ecommercee/repository/OrderItemRepository.java]: To be agreed with others. You know my opinion :)

**Review by mwojtyla — COMMENTED** [review]
We can create tests in many ways. Your version looks good :)
Just a few minor comments from me.

File: build.gradle (hunk 1 above)
Comment (WitoZak) [reply build.gradle]: Junit

I'm not reinventing the wheel, I took it from the Jetbrains video about setting up Junit5 for Gradle in Intellij.

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 2 above)
Comment (WitoZak) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: The '@NotNull' annotation imposes restrictions on the fields of the class, which means that the no-argument constructor generated by the '@NoArgsConstructor' annotation will not initialize these fields. To avoid NullPointerException errors, you should add (force = true), which enforces the generation of a constructor for fields with the '@NotNull' annotation.
NotNull

File: build.gradle (hunk 1 above)
Comment (mwojtyla) [reply build.gradle]: You don't have to be rude :( I just wrote that it's not necessary to change that. The version we have works, so it's just another way which is correct...

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 2 above)
Comment (mwojtyla) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: Hmm, we didn't need to use it before. I think we should asked Andrzej to explain that.

File: build.gradle (hunk 1 above)
Comment (WitoZak) [reply build.gradle]: I understand, I didn't mean to come off as rude 😃  . As far as I'm concerned, we're still learning and something new is always appearing. We should update the things we use. To show you that I'm not being unfriendly, we'll do it your way 👍

File: src/main/java/com/kodilla/ecommercee/repository/OrderItemRepository.java (hunk 4 above)
Comment (WitoZak) [reply src/main/java/com/kodilla/ecommercee/repository/OrderItemRepository.java]: For your peace of mind, we will do it your way.

File: build.gradle (hunk 1 above)
Comment (mwojtyla) [reply build.gradle]: It's not "my way" - someone made that in our project :) Mayby you are correct and it's wrong. I don't know. As you wrote, we're still learning. It's good that you found that difference, but if we explore the topic, I wish to know why do we need to change it? How does this affect the code? Is it just a recording issue?

**Review by mwojtyla — APPROVED** [review]

**Review by KamilBonar — APPROVED** [review]

**Review by michal2220 — APPROVED** [review]
Looks ok for me :)
